docs/overview.rst

```diff
@@ -0,0 +1,350 @@
+if you want to write an extension in a language different than C.
+
+However, Python bindings for other languages could decide to target the HPy
+Universal ABI instead of the CPython ABI, and generate extensions which can be
+loaded seamlessly on all Python implementations which supports it.  This is
+the route taken for example by Rust (XXX put a link?).
+
+
+Benefits for alternative Python implementations
+------------------------------------------------
+
+If you are writing an alternative Python implementation, there is a good
+chance that you already know how painful it is to support the Python/C
+API. HPy is designed to be both faster and easier to implement!
+
+You have two choices:
+
+  - support the Universal ABI: in this case, you just need to export the
+    needed functions and to add a hook to ``dlopen()`` the desired libraries
+
+  - use a custom ABI: in this case, you have to write your own replacement for
+    ``hpy.h`` and recompile the C extensions with it.
+
+
+Current status and roadmap
+--------------------------
+
+At the moment of writing, HPy is still in its early stages of development. The
+following milestones have been reached:
+
+  - it is possible to write extensions which expose module-level functions,
+    with all the various kinds of calling conventions
+
+  - there is a limited support for argument parsing (only a couple of basic
+    types actually work)
+
+  - there is support for raising and catching exceptions
+
+  - it is possible to choose between the CPython and the HPy Universal ABIs
+    when compiling an extension module
+
+  - extensions compiled with the CPython ABI work out of the box on CPython
+
+  - it is possible to load HPy Universal extensions on CPython, thanks to the
+    ``hpy.universal`` package
+
+  - it is possible to load HPy Universal extensions on
+    PyPy (using the PyPy `hpy branch <https://foss.heptapod.net/pypy/pypy/tree/branch/hpy>`_
+
+  - it is possible to load HPy Universal extensions on GraalPython (XXX link)
+
+
+However, there is still a long road before HPy is usable for the general
+public. In particular, the following features are on our roadmap but has not
+been implemented yet:
+
+  - it is not possible to write custome types in C. There is already a WIP
+    branch to address this issue
+
+  - only a handful of the original Python/C functions has been ported to
+    HPy. Porting most of them is straighforward, so for now the priority is to
+    work on the "hard" features to prove that the HPy approach works, and we
+    port new functions as needed
+
+  - the debug mode simply does not exist (yet!)
+
+  - there is not a standard/easy way to integrate HPy in a
+    ``distutils/setuptools`` based workflow. The only way to compile HPy
+    extensions right now is to manually close the git repo and tweak your
+    ``setup.py``. Eventually, we plan to offer a workflow which integrates
+    seamlessly with ``pip``, ``setuptools``, etc.
```

Comment:
Worth thinking about [PEP 517](https://www.python.org/dev/peps/pep-0517/) type of build environment (not for this doc, but something to think about)

Reply:
I didn't know PEP 517. Yes, it's worth thinking about it